### PR TITLE
Improve docker documentation

### PR DESCRIPTION
**Pull request by alekzvik**
- Move out docker docs to a separate file.
- Add an explanation of how it works internally
- Add `FAQ` section

Interest was expressed here: #511

### CODE REVIEW

**Comment by alekzvik** [comment]
@m-mohr since you first expressed interest in this doc, please take a look and let me know what is missing and what can be improved

File: docs/docker.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Docker details
+
+Note: Docker might not be an ideal way to deploy stac browser in production. Consider using cloud storage and CDN.
```
Comment (m-mohr) [inline docs/docker.md:3]:
```suggestion
Note: Docker might not be an ideal way to deploy STAC Browser in production. Consider using a web host, cloud storage, or a CDN. 
```

File: docs/docker.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Docker details
+
+Note: Docker might not be an ideal way to deploy stac browser in production. Consider using cloud storage and CDN. 
+
+# How it works
```
Comment (m-mohr) [inline docs/docker.md:5]:
```suggestion
## How it works

```

File: docs/docker.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+The docker image uses multi stage build.
+The first stage is based on a node image and runs `npm build` to produce a /dist dir with static files.
+The second stage is based on nginx image that serves the folder with static files and deals with `pathPrefix` arg.
+So, essentialy, in the end you get an nginx instance that serves static files.
+
+# Essential parts
```
Comment (m-mohr) [inline docs/docker.md:11]:
```suggestion
## Essential parts

```

File: docs/docker.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Docker details
+
+Note: Docker might not be an ideal way to deploy stac browser in production. Consider using cloud storage and CDN. 
+
+# How it works
+The docker image uses multi stage build.
+The first stage is based on a node image and runs `npm build` to produce a /dist dir with static files.
+The second stage is based on nginx image that serves the folder with static files and deals with `pathPrefix` arg.
+So, essentialy, in the end you get an nginx instance that serves static files.
```
Comment (m-mohr) [inline docs/docker.md:9]:
```suggestion
The docker image uses a multi stage build.
The first stage is based on a node image and runs `npm build` to produce a `/dist` folder with static files (HTML, CSS, and JavaScript).
The second stage is based on a nginx image that serves the folder with static files and deals with the build-only options such as `pathPrefix`.
So, essentialy, in the end you get an nginx instance that serves the static files.
```

File: docs/docker.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+# Essential parts
+1. Dockerfile - contains information on how to build the image.
+2. /docker/default.conf - nginx configuration template, where `<pathPrefix>` var is replaced during build.
+3. /docker/docker-entrypoint.sh - a start script to read the passed variables and produce `config.js` file.
```
Comment (m-mohr) [inline docs/docker.md:14]:
```suggestion
1. [Dockerfile](../Dockerfile) - contains information on how to build the image.
2. [docker/default.conf](../docker/default.conf) - nginx configuration template, where `<pathPrefix>` is replaced during build.
3. [docker/docker-entrypoint.sh](../docker/docker-entrypoint.sh) - a start script to read the passed variables and produce the `config.js` file.
```

File: docs/docker.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+# FAQ
```
Comment (m-mohr) [inline docs/docker.md:16]:
```suggestion
## FAQ

```

File: docs/docker.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+> Can I use `ghcr.io/radiantearth/stac-browser` image with the `pathPrefix`?
+
+You can not. You need to build your own image because `pathPrefix` is a build only param.
```
Comment (m-mohr) [inline docs/docker.md:19]:
```suggestion
You can not. You need to build your own image because `pathPrefix` is a build-only option.
```

**Review by m-mohr — CHANGES_REQUESTED** [review]
Thanks, I appreciate it. I left a couple of editorial suggestions.

Generally, I think we could move the whole Docker section from the README to the docker.md and just leave the header and a link to the docs in the README. That would slim down the README a bit. But just an idea, not a must.

**Comment by alekzvik** [comment]
I agree with your suggestion to move out the docker section. 
I yanked it in full in this revision and left a small note with a link in README.
Other editorial changes are also fixed.

File: docs/docker.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+services:
+  stac-browser:
+    image: ghcr.io/radiantearth/stac-browser:latest
+    ports:
+      - 8080:8080
+    environment:
+      SB_catalogUrl: "https://localhost:7188"
+```
+
+## How it works
```
Comment (m-mohr) [inline docs/docker.md:64]:
```suggestion
## How it works

```

File: docs/docker.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+The docker image uses a multi stage build.
+The first stage is based on a node image and runs `npm build` to produce a `/dist` folder with static files (HTML, CSS, and JavaScript).
+The second stage is based on an nginx image that serves the folder with static files and deals with the build-only options such as  `pathPrefix`.
+So, essentialy, in the end you get an nginx instance that serves static files.
+
+## Essential parts
```
Comment (m-mohr) [inline docs/docker.md:70]:
```suggestion
## Essential parts

```

File: docs/docker.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+1. [Dockerfile](../Dockerfile) - contains information on how to build the image.
+2. [docker/default.conf](../docker/default.conf) - nginx configuration template, where `<pathPrefix>` is replaced during build.
+3. [docker/docker-entrypoint.sh](../docker/docker-entrypoint.sh) - a start script to read the passed variables and produce the `config.js` file.
+
+## FAQ
```
Comment (m-mohr) [inline docs/docker.md:75]:
```suggestion
## FAQ

```

**Review by m-mohr — APPROVED** [review]
Thank you very much!

**Comment by alekzvik** [comment]
@m-mohr Thank you for all the corrections and editing.

**Pull request merged**